### PR TITLE
clone repo when adding package using the giturl instead of using the api

### PR DESCRIPTION
**Pull request by ahelsayd**
see https://github.com/threefoldtech/js-sdk/issues/35

### CODE REVIEW

File: jumpscale/servers/threebot/threebot.py [hunk 1]
```diff
@@ -241,13 +234,21 @@ def list_all(self):
     def add(self, path: str = None, giturl: str = None):
         # TODO: Check if package already exists
 
-        if any([path, giturl]) and not all([path, giturl]):
+        if not any([path, giturl]) or all([path, giturl]):
```
Comment (xmonader) [inline jumpscale/servers/threebot/threebot.py:237]: should be `if not any([path, giturl]) and not all([path, giturl])`

we want path or giturl to be there but not both of them

File: jumpscale/servers/threebot/threebot.py [hunk 2]
```diff
@@ -241,13 +234,21 @@ def list_all(self):
...
             raise j.exceptions.Value("either path or giturl is required")
 
         if giturl:
-            org, repo, _, branch, path = urlparse(giturl).path.lstrip("/").split("/", 4)
-            repo = self.github.get_repo(f"{org}/{repo}")
-            path = repo.download_directory(path, DOWNLOADED_PACKAGES_PATH, branch=branch)
+            url = urlparse(giturl)
+            org, repo, _, branch, package_path = url.path.lstrip("/").split("/", 4)
```
Comment (xmonader) [inline jumpscale/servers/threebot/threebot.py:242]: can we validate the number of components before proceeding?

File: jumpscale/servers/threebot/threebot.py (hunk 1 above)
Comment (waleedhammam) [reply jumpscale/servers/threebot/threebot.py:237]: we need the condition to be the opposite, if it enteres the if it'll raise not continue

thanks @abom

Comment (ahelsayd) [reply jumpscale/servers/threebot/threebot.py:237]: i think it is correct
raise error if  not any of path / giturl passed or if both of them are passed

**Pull request merged**